src/js/components/Menu.js

```diff
@@ -107,19 +111,50 @@ var Menu = React.createClass({
 
   _onOpen: function (event) {
     event.preventDefault();
-    this.setState({active: true});
+    this.setState({state: 'expanded'});
   },
 
   _onClose: function () {
-    this.setState({active: false});
+    this.setState({state: 'collapsed'});
+    if (document.activeElement === this.getDOMNode()) {
+      this.setState({state: 'focused'});
+    } else {
+      this.getDOMNode().focus();
+    }
+  },
+
+  _onUpKeyPress: function (event) {
+    if (!this._drop.activeMenuItem) {
+      var menuItems = document.getElementById(this._navContainerId).childNodes;
+      var lastMenuItem = menuItems[menuItems.length - 1];
+      this._drop.activeMenuItem = lastMenuItem;
+    } else if (this._drop.activeMenuItem.previousSibling) {
+      this._drop.activeMenuItem = this._drop.activeMenuItem.previousSibling;
+    }
+
+    this._drop.activeMenuItem.focus();
+    return true;
+  },
+
+  _onDownKeyPress: function (event) {
+    if (!this._drop.activeMenuItem) {
+      this._drop.activeMenuItem = document.getElementById(this._navContainerId).childNodes[0];
+    } else if (this._drop.activeMenuItem.nextSibling) {
+      this._drop.activeMenuItem = this._drop.activeMenuItem.nextSibling;
+    }
+
+    this._drop.activeMenuItem.focus();
+    return true;
```

Comment:
can we add a comment that explain why `return true` is required here? I remember it is for stopPropagation, but it is good to explain it ;)